File: 2023/day01-1.js
import fs from "fs";

const data = fs.readFileSync("./data/input.txt", { encoding: "utf-8" });
const lines = data.split("\n");

let sum = 0;
for (const line of lines) {
  const matches = line.match(/\d+/g);
  const joined = matches.join("");
  const first = joined[0];
  const last = joined[joined.length - 1];
  const number = parseInt(`${first}${last}`);
  sum += number;
}
console.log(sum);

File: 2023/day01-1.go
package main

import (
	"fmt"
	"os"
	"regexp"
	"strconv"
	"strings"
)

func main() {
	content, _ := os.ReadFile("./data/input.txt")
	lines := strings.Split(string(content), "\n")
	r, _ := regexp.Compile(`\d`)
	total := 0
	for _, line := range lines {
		matches := r.FindAllString(line, -1)
		fmt.Println(matches)
		first := matches[0]
		last := matches[len(matches)-1]
		combined := fmt.Sprintf("%s%s", first, last)
		sum, _ := strconv.Atoi(combined)
		fmt.Println(sum)
		total += sum
	}
	fmt.Println(total)
}
